spec/index.bs

```diff
@@ -6518,9 +6518,97 @@ and link it together with the shaders provided by the user.
 This linking is one of the reason the object is created as a whole.
 
 This combination state is created as a single object
-(by {{GPUDevice/createComputePipeline(descriptor)|GPUDevice.createComputePipeline()}} or {{GPUDevice/createRenderPipeline(descriptor)|GPUDevice.createRenderPipeline()}}),
-and switched as one
-(by {{GPUComputePassEncoder/setPipeline(pipeline)|GPUComputePassEncoder.setPipeline}} or {{GPURenderCommandsMixin/setPipeline(pipeline)|GPURenderCommandsMixin.setPipeline}} correspondingly).
+(a {{GPUComputePipeline}} or {{GPURenderPipeline}})
+and switched using one command
+({{GPUComputePassEncoder}}.{{GPUComputePassEncoder/setPipeline()}} or
+{{GPURenderCommandsMixin}}.{{GPURenderCommandsMixin/setPipeline()}} respectively).
+
+There are two ways to create pipelines:
+
+: <dfn dfn>immediate pipeline creation</dfn>
+:: {{GPUDevice/createComputePipeline()}} and {{GPUDevice/createRenderPipeline()}}
+    return a pipeline object which can be used immediately in a pass encoder.
+
+    When this fails, the pipeline object will be invalid and the call will generate either a
+    [$validation error$] or an [$internal error$].
+
+    Note: If pipeline creation takes a long time, this can incur a stall in the
+    [=device timeline=] at some point between the creation call and execution of the
+    {{GPUQueue/submit()}} in which it is first used.
+    The point is unspecified, but most likely to be one of: at creation, at the first usage of the
+    pipeline in `setPipeline()`, at the corresponding `finish()` of that {{GPUCommandEncoder}} or
+    {{GPURenderBundleEncoder}}, or at {{GPUQueue/submit()}} of that {{GPUCommandBuffer}}.
+
+: <dfn dfn>promised pipeline creation</dfn>
```

Comment:
Non-blocking Nit: I get why you probably don't want to use the terms "deferred" or "asynchronous" here, but "promised" sounds strange, especially when the method itself if called `*Async`.

IMO if we're using "immediate" as the term above (even though it's actually async and just returns a result immediately) then it'd be OK to use the term "asynchronous" here as it matches the return behavior.